Index: ExtensionLoader.csproj
===================================================================
--- ExtensionLoader.csproj	(revision 6)
+++ ExtensionLoader.csproj	(working copy)
@@ -33,11 +33,12 @@
     <DebugSymbols>True</DebugSymbols>
     <FileAlignment>4096</FileAlignment>
     <Optimize>False</Optimize>
-    <OutputPath>bin\</OutputPath>
+    <OutputPath>..\bin\</OutputPath>
     <RegisterForComInterop>False</RegisterForComInterop>
     <RemoveIntegerChecks>False</RemoveIntegerChecks>
     <TreatWarningsAsErrors>False</TreatWarningsAsErrors>
     <WarningLevel>4</WarningLevel>
+    <NoStdLib>False</NoStdLib>
     <NoWarn>1591,1574,0419,0618</NoWarn>
   </PropertyGroup>
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
@@ -51,17 +52,17 @@
     <DebugSymbols>False</DebugSymbols>
     <FileAlignment>4096</FileAlignment>
     <Optimize>True</Optimize>
-    <OutputPath>bin\</OutputPath>
+    <OutputPath>..\bin\</OutputPath>
     <RegisterForComInterop>False</RegisterForComInterop>
     <RemoveIntegerChecks>False</RemoveIntegerChecks>
     <TreatWarningsAsErrors>False</TreatWarningsAsErrors>
     <WarningLevel>4</WarningLevel>
+    <NoStdLib>False</NoStdLib>
     <NoWarn>1591,1574,0419,0618</NoWarn>
   </PropertyGroup>
   <ItemGroup>
     <Reference Include="System">
         <Name>System</Name>
-      <HintPath>\System.dll</HintPath>
     </Reference>
   </ItemGroup>
   <ItemGroup>
@@ -73,94 +74,94 @@
     <Compile Include="IExtension.cs">
       <SubType>Code</SubType>
     </Compile>
-    <Compile Include="Config\AliasText.cs">
+    <Compile Include="Ini\IniException.cs">
       <SubType>Code</SubType>
     </Compile>
-    <Compile Include="Config\ConfigBase.cs">
+    <Compile Include="Ini\IniWriter.cs">
       <SubType>Code</SubType>
     </Compile>
-    <Compile Include="Config\ConfigCollection.cs">
+    <Compile Include="Ini\IniDocument.cs">
       <SubType>Code</SubType>
     </Compile>
-    <Compile Include="Config\ConfigSourceBase.cs">
+    <Compile Include="Ini\IniItem.cs">
       <SubType>Code</SubType>
     </Compile>
-    <Compile Include="Config\IConfig.cs">
+    <Compile Include="Ini\IniSectionCollection.cs">
       <SubType>Code</SubType>
     </Compile>
-    <Compile Include="Config\IConfigSource.cs">
+    <Compile Include="Ini\IniReader.cs">
       <SubType>Code</SubType>
     </Compile>
-    <Compile Include="Config\IniConfig.cs">
+    <Compile Include="Ini\IniSection.cs">
       <SubType>Code</SubType>
     </Compile>
-    <Compile Include="Config\IniConfigSource.cs">
+    <Compile Include="Config\IConfigSource.cs">
       <SubType>Code</SubType>
     </Compile>
-    <Compile Include="Ini\IniDocument.cs">
+    <Compile Include="Config\IConfig.cs">
       <SubType>Code</SubType>
     </Compile>
-    <Compile Include="Ini\IniException.cs">
+    <Compile Include="Config\AliasText.cs">
       <SubType>Code</SubType>
     </Compile>
-    <Compile Include="Ini\IniItem.cs">
+    <Compile Include="Config\ConfigCollection.cs">
       <SubType>Code</SubType>
     </Compile>
-    <Compile Include="Ini\IniReader.cs">
+    <Compile Include="Config\ConfigBase.cs">
       <SubType>Code</SubType>
     </Compile>
-    <Compile Include="Ini\IniSection.cs">
+    <Compile Include="Config\ConfigSourceBase.cs">
       <SubType>Code</SubType>
     </Compile>
-    <Compile Include="Ini\IniSectionCollection.cs">
+    <Compile Include="Config\IniConfig.cs">
       <SubType>Code</SubType>
     </Compile>
-    <Compile Include="Ini\IniWriter.cs">
+    <Compile Include="Config\IniConfigSource.cs">
       <SubType>Code</SubType>
     </Compile>
-    <Compile Include="Nini\Config\AliasText.cs">
+    <Compile Include="Nini\Ini\IniException.cs">
       <SubType>Code</SubType>
     </Compile>
-    <Compile Include="Nini\Config\ConfigBase.cs">
+    <Compile Include="Nini\Ini\IniWriter.cs">
       <SubType>Code</SubType>
     </Compile>
-    <Compile Include="Nini\Config\ConfigCollection.cs">
+    <Compile Include="Nini\Ini\IniDocument.cs">
       <SubType>Code</SubType>
     </Compile>
-    <Compile Include="Nini\Config\ConfigSourceBase.cs">
+    <Compile Include="Nini\Ini\IniItem.cs">
       <SubType>Code</SubType>
     </Compile>
-    <Compile Include="Nini\Config\IConfig.cs">
+    <Compile Include="Nini\Ini\IniReader.cs">
       <SubType>Code</SubType>
     </Compile>
-    <Compile Include="Nini\Config\IConfigSource.cs">
+    <Compile Include="Nini\Ini\IniSectionCollection.cs">
       <SubType>Code</SubType>
     </Compile>
-    <Compile Include="Nini\Config\IniConfig.cs">
+    <Compile Include="Nini\Ini\IniSection.cs">
       <SubType>Code</SubType>
     </Compile>
-    <Compile Include="Nini\Config\IniConfigSource.cs">
+    <Compile Include="Nini\Config\IConfigSource.cs">
       <SubType>Code</SubType>
     </Compile>
-    <Compile Include="Nini\Ini\IniDocument.cs">
+    <Compile Include="Nini\Config\IConfig.cs">
       <SubType>Code</SubType>
     </Compile>
-    <Compile Include="Nini\Ini\IniException.cs">
+    <Compile Include="Nini\Config\AliasText.cs">
       <SubType>Code</SubType>
     </Compile>
-    <Compile Include="Nini\Ini\IniItem.cs">
+    <Compile Include="Nini\Config\ConfigSourceBase.cs">
       <SubType>Code</SubType>
     </Compile>
-    <Compile Include="Nini\Ini\IniReader.cs">
+    <Compile Include="Nini\Config\ConfigBase.cs">
       <SubType>Code</SubType>
     </Compile>
-    <Compile Include="Nini\Ini\IniSection.cs">
+    <Compile Include="Nini\Config\ConfigCollection.cs">
       <SubType>Code</SubType>
     </Compile>
-    <Compile Include="Nini\Ini\IniSectionCollection.cs">
+    <Compile Include="Nini\Config\IniConfig.cs">
       <SubType>Code</SubType>
     </Compile>
-    <Compile Include="Nini\Ini\IniWriter.cs">
+    <Compile Include="Nini\Config\IniConfigSource.cs">
       <SubType>Code</SubType>
     </Compile>
     <Compile Include="Nini\Util\OrderedList.cs">
