Index: Extensions/CapsManager.cs
===================================================================
--- Extensions/CapsManager.cs	(revision 2416)
+++ Extensions/CapsManager.cs	(working copy)
@@ -6,6 +6,7 @@
 using OpenMetaverse;
 using OpenMetaverse.Http;
 using OpenMetaverse.StructuredData;
+using Logger = OpenMetaverse.Logger;
 
 namespace Simian.Extensions
 {
Index: Extensions/SceneManager.cs
===================================================================
--- Extensions/SceneManager.cs	(revision 2416)
+++ Extensions/SceneManager.cs	(working copy)
@@ -432,6 +432,10 @@
             appearance.ObjectData.TextureEntry = agent.Avatar.Textures.ToBytes();
             appearance.Sender.ID = agent.Avatar.ID;
             appearance.Sender.IsTrial = false;
+            if (agent.VisualParams == null)
+            {
+                agent.VisualParams = new byte[0];
+            }
 
             appearance.VisualParam = new AvatarAppearancePacket.VisualParamBlock[agent.VisualParams.Length];
             for (int i = 0; i < agent.VisualParams.Length; i++)
Index: Extensions/Periscope.cs
===================================================================
--- Extensions/Periscope.cs	(revision 2416)
+++ Extensions/Periscope.cs	(working copy)
@@ -9,7 +9,7 @@
 {
     public class Periscope : IExtension<Simian>
     {
-        public Agent MasterAgent = null;
+       public Agent MasterAgent = null;
 
         Simian server;
         GridClient client;
@@ -26,8 +26,9 @@
         {
             this.server = server;
 
-            client = new GridClient();
-            Settings.LOG_LEVEL = Helpers.LogLevel.Info;
+            //client = new GridClient();
+            client = server.periscopeClient;
+            Settings.LOG_LEVEL = Helpers.LogLevel.Debug;
             client.Settings.MULTIPLE_SIMS = false;
             client.Settings.SEND_AGENT_UPDATES = false;
 
@@ -103,8 +104,9 @@
             server.Scene.ObjectTransform(this, update.LocalID, update.Position, update.Rotation, update.Velocity,
                 update.Acceleration, update.AngularVelocity);
 
-            if (update.LocalID == client.Self.LocalID)
+            if (update.LocalID == client.Self.LocalID && MasterAgent!=null)
             {
+                
                 MasterAgent.Avatar.Acceleration = update.Acceleration;
                 MasterAgent.Avatar.AngularVelocity = update.AngularVelocity;
                 MasterAgent.Avatar.CollisionPlane = update.CollisionPlane;
Index: Simian.cs
===================================================================
--- Simian.cs	(revision 2416)
+++ Simian.cs	(working copy)
@@ -11,11 +11,12 @@
 using HttpServer;
 using OpenMetaverse;
 using OpenMetaverse.Http;
-
+using Logger = OpenMetaverse.Logger;
 namespace Simian
 {
     public partial class Simian
     {
+        public GridClient periscopeClient;
         public const string CONFIG_FILE = "Simian.ini";
 
         // TODO: Don't hard-code these
@@ -65,7 +66,7 @@
             }
             catch (Exception)
             {
-                Logger.Log("Failed to load [Extensions] section from " + CONFIG_FILE, Helpers.LogLevel.Error);
+                OpenMetaverse.Logger.Log("Failed to load [Extensions] section from " + CONFIG_FILE, Helpers.LogLevel.Error);
                 return false;
             }
 
@@ -77,9 +78,9 @@
             {
                 // Load all of the extensions
                 List<string> references = new List<string>();
-                references.Add("OpenMetaverseTypes.dll");
-                references.Add("OpenMetaverse.dll");
-                references.Add("Simian.exe");
+               // references.Add("OpenMetaverseTypes.dll");
+               // references.Add("OpenMetaverse.dll");
+                references.Add("superCogbot.exe");
 
                 List<FieldInfo> assignables = ExtensionLoader<Simian>.GetInterfaces(this);
 
@@ -104,7 +105,7 @@
             // Start all of the extensions
             foreach (IExtension<Simian> extension in ExtensionLoader<Simian>.Extensions)
             {
-                Logger.Log("Starting extension " + extension.GetType().Name, Helpers.LogLevel.Info);
+                OpenMetaverse.Logger.Log("Starting extension " + extension.GetType().Name, Helpers.LogLevel.Info);
                 extension.Start(this);
             }
 
@@ -258,7 +259,7 @@
             }
             catch (Exception ex)
             {
-                Logger.Log("XmlRpc login error: " + ex.Message, Helpers.LogLevel.Error, ex);
+                OpenMetaverse.Logger.Log("XmlRpc login error: " + ex.Message, Helpers.LogLevel.Error, ex);
             }
 
             return true;
Index: Main.cs
===================================================================
--- Main.cs	(revision 2416)
+++ Main.cs	(working copy)
@@ -6,7 +6,7 @@
 {
     class MainEntry
     {
-        static void Main(string[] args)
+        static void SimianMain(string[] args)
         {
             Simian simulator = new Simian();
             if (simulator.Start())
